packages/data-schema/__tests__/__snapshots__/CustomOperations.test.ts.snap

```diff
@@ -1,5 +1,201 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
+exports[`CustomOperation transform custom operations with custom types and refs Schema with custom mutation using custom type argument 1`] = `
+"input MutateWithCustomTypeArgCustomArgInput
+{
+  field: String
+}
+
+type CustomArgType 
+{
+  field: String
+}
+
+type Mutation {
+  mutateWithCustomTypeArg(customArg: MutateWithCustomTypeArgCustomArgInput): String @function(name: "myFunc") @auth(rules: [{allow: public, provider: apiKey}])
+}"
+`;
+
+exports[`CustomOperation transform custom operations with custom types and refs Schema with custom mutation using enum ref in nested custom type 1`] = `
+"enum SomeEnum {
+  VALUE1
+  VALUE2
+}
+
+input MutateWithNestedEnumRefNestedArgInput
+{
+  field: String
+  enumField: SomeEnum
+}
+
+type NestedCustomType 
+{
+  field: String
+  enumField: SomeEnum
+}
+
+type Mutation {
+  mutateWithNestedEnumRef(nestedArg: MutateWithNestedEnumRefNestedArgInput): String @function(name: "myFunc") @auth(rules: [{allow: public, provider: apiKey}])
+}"
+`;
+
+exports[`CustomOperation transform custom operations with custom types and refs Schema with custom mutation using ref argument 1`] = `
+"input MutationWithRefArgRefArgInput
+{
+  field: String
+}
+
+type RefArgType 
+{
+  field: String
+}
+
+type Mutation {
+  mutationWithRefArg(refArg: MutationWithRefArgRefArgInput): String @function(name: "myFunc") @auth(rules: [{allow: public, provider: apiKey}])
+}"
+`;
+
+exports[`CustomOperation transform custom operations with custom types and refs Schema with custom operation using enum directly in arguments 1`] = `
+"enum DirectEnumQueryStatus {
+  PENDING
+  APPROVED
+  REJECTED
+}
+
+type Query {
+  directEnumQuery(status: DirectEnumQueryStatus): String @function(name: "myFunc") @auth(rules: [{allow: public, provider: apiKey}])
+}"
+`;
+
+exports[`CustomOperation transform custom operations with custom types and refs Schema with custom operation using enum in inline custom type arguments 1`] = `
+"input InlineEnumQueryInputInput
+{
+  name: String
+  status: InlineEnumQueryInputInputStatus
+}
+
+enum InlineEnumQueryInputInputStatus {
+  ACTIVE
+  INACTIVE
+}
+
+type Query {
+  inlineEnumQuery(input: InlineEnumQueryInputInput): String @function(name: "myFunc") @auth(rules: [{allow: public, provider: apiKey}])
+}"
+`;
+
+exports[`CustomOperation transform custom operations with custom types and refs Schema with custom operation using enum in referenced custom type arguments 1`] = `
+"input RefEnumQueryInputInput
```

Comment:
I think this should be `StatusInputInput`. In this case, maybe the root type could be named `StatusType` leaving this `StatusTypeInput`